core/src/main/java/com/squarespace/template/plugins/platform/ContentFormatters.java

```diff
@@ -378,22 +379,27 @@ public void apply(Context ctx, Arguments args, Variables variables) throws CodeE
     }
 
     private String getAltText(Context ctx) {
-      // For image blocks, caption is stored on the block and not the item.
-      // need to reach out via the context to see if it exist first,
-      // before falling back on the data on the item
-
-      // this will be empty if this is not a block
-      JsonNode blockInfo = ctx.resolve("info");
-      if (blockInfo != null) {
-        String altText = StringUtils.trimToNull(blockInfo.path("altText").asText());
-        if (altText != null) {
-          return altText;
-        }
+      // Content Items for image blocks were populated with an altText value with a migration. See CMS-33805.
+      // The contentItem value should always be used, even if it is empty.
+      JsonNode blockType = ctx.resolve("blockType");
+
+      if (!blockType.isMissingNode() && blockType.textValue().toLowerCase().equals("image")) {
+        return getAltTextFromImageBlock(ctx);
       }
 
       JsonNode image = ctx.node();
       return getAltTextFromContentItem(image);
     }
+
+    private String getAltTextFromImageBlock(Context ctx) {
```

Comment:
It looks like this method could have a more generic name, right? We'll be reusing it for other content item types when we migrate those, not just for image blocks.

Reply:
Pat's rewrite of this method was only two lines, so I removed the method and just moved the code into `getAltText()`.